src/algorithm/centroid.rs

```diff
@@ -6,6 +6,8 @@ use algorithm::distance::Distance;
 
 /// Calculation of the centroid.
 pub trait Centroid<T: Float> {
+    type Centroid;
```

Comment:
actually, maybe `Output` would be a better name here. i'm doing the same thing with bounding box right now and it's confusing that the trait is `BoundingBox`, the associated return type name is `BoundingBox` and the actual return type is `Bbox`.